id: unauth-jupyter-lab

info:
  name: Jupyter Lab - Unauthenticated Access
  author: j4vaovo
  severity: critical
  description: |
    JupyterLab was able to be accessed without authentication.
  reference:
    - https://paper.seebug.org/2058/
  classification:
    cvss-metrics: CVSS:3.0/AV:N/AC:L/PR:N/UI:N/S:C/C:H/I:H/A:H
    cvss-score: 10
    cwe-id: CWE-288
  metadata:
    max-request: 1
    shodan-query: http.favicon.hash:450899026
  tags: unauth,jupyter,jupyterlab,misconfig

http:
  - method: GET
    path:
      - '{{BaseURL}}/lab/api/settings/'

    matchers-condition: and
    matchers:
      - type: word
        part: body
        words:
          - '"id":'
          - '"settings"'
          - '"schema":'
        condition: and

      - type: word
        part: header
        words:
          - "application/json"

      - type: status
        status:
          - 200
# digest: 4a0a004730450220200f9033158fb08bb67baa53e6bdfe88e6495b276f5553d9608dc33a53183d45022100b3fd5ed1bcdac1435c230111079866d8f8653bf748e1f5a968dcda55ad6dc842:922c64590222798bb761d5b6d8e72950